autodistill/cli.py

```diff
@@ -0,0 +1,167 @@
+):
+    if dataset_format not in SUPPORTED_DATASET_FORMATS:
+        print(
+            "Dataset format not supported. Please choose from the following dataset formats: "
+            + str(SUPPORTED_DATASET_FORMATS)
+        )
+        exit()
+
+    if model_type not in SUPPORTED_MODEL_TYPES:
+        print(
+            "Model type not supported. Please choose from the following model types: "
+            + str(SUPPORTED_MODEL_TYPES)
+        )
+        exit()
+
+    if ontology == "{}":
+        print("No ontology provided. Please provide an ontology.")
+        exit()
+
+    print("Loading base model...")
+    model = import_requisite_module(base)
+
+    try:
+        ontology = json.loads(ontology, strict=False)
+    except:
+        print("Ontologies must be valid JSON.")
+        exit()
+        
+    if upload_to_roboflow:
+        roboflow.login()
+
+        if model_value not in SUPPORTED_ROBOFLOW_MODEL_UPLOADS:
+            print(
+                f"Model type {model_value} is not supported by Roboflow. Please choose one of {SUPPORTED_ROBOFLOW_MODEL_UPLOADS}."
+            )
+            exit()
+
+    ontology = CaptionOntology(ontology)
+
+    base_model = model(ontology=ontology)
+
+    print("Labeling data...")
+
+    # if dir is file, run label on file
+    if os.path.isfile(images):
+        classes = ontology.classes()
+
+        box_annotator = sv.BoxAnnotator()
+
+        image = cv2.imread(images)
+
+        base_model.label(
+            input_folder=os.path.dirname(images),
+            extension=os.path.splitext(images)[1],
+            output_folder=output,
+        )
+        # show results
+        detections = base_model.predict(images)
+
+        labels = [
+            f"{classes[class_id]} {confidence:0.2f}"
+            for _, _, confidence, class_id, _ in detections
+        ]
+
+        annotated_frame = box_annotator.annotate(
+            scene=image.copy(),
+            detections=detections,
+            labels=labels,
+        )
+
+        sv.plot_image(annotated_frame, size=(8, 8))
+
+        print("Saving results to ./result.jpg...")
+        
+        cv2.imwrite(os.path.join(os.getcwd(), "result.jpg"), annotated_frame)
+
+        exit()
+
+    base_model.label(
+        input_folder=dir,
+        output_folder=output
+    )
+
+    print("Loading target model...")
+    target_model = import_requisite_module(target)
+
+    print("Training target model...")
+    target_model.train(dataset_yaml=os.path.join(output, "data.yaml"), epochs=epochs)
+
+    if upload_to_roboflow:
+        rf = roboflow.Roboflow()
+
+        workspace = rf.workspace()
+
+        if model_type == "detection":
+            rf_model_value = "object-detection"
+        elif model_type == "segmentation":
+            rf_model_value = "image-segmentation"
+        else:
+            rf_model_value = "single-label-classification"
+
+        project = workspace.create_project(
+            project_name=upload_to_roboflow,
```

Comment:
What happens if they pass `True`? Does this still setup a good default name for them?

Reply:
I have updated this so that the default is `""`, which is falsy in Python. Therefore, you must specify a project name for a project to be uploaded to Roboflow.